mmengine/utils/package_utils.py

```diff
@@ -45,15 +43,31 @@ def get_installed_path(package: str) -> str:
     """
     import importlib.util
 
-    from pkg_resources import DistributionNotFound, get_distribution
-
     # if the package name is not the same as module name, module name should be
     # inferred. For example, mmcv-full is the package name, but mmcv is module
     # name. If we want to get the installed path of mmcv-full, we should concat
     # the pkg.location and module name
     try:
-        pkg = get_distribution(package)
-    except DistributionNotFound as e:
+        dist = distribution(package)
+        # In importlib.metadata, we use dist.locate_file() or files
+        if hasattr(dist, 'locate_file'):
```

Comment:
This time, we consider bumping the minimum python version to 3.10, just keep the locate file branch.